src/oauth-backend.js

```diff
@@ -156,60 +171,60 @@ export default class OAuthBackend extends AuthBackend {
 				}
 			}
 		}
-
-		if (!passive) {
-			await this.activeLogin(rest);
-		}
-
-		if (this.isAuthenticated()) {
-			let user = await this.getUser();
-			this.dispatchEvent(new CustomEvent("mv-login"));
-			this.updatePermissions({login: false, logout: true});
-			return user;
-		}
 	}
 
-	async activeLogin () {
-		let oAuthBackend = this.constructor.getOAuthBackend();
+	async activeLogin (o) {
+		let accessToken;
 
-		// Show window
-		let popup = {
-			width: Math.min(1000, innerWidth - 100),
-			height: Math.min(800, innerHeight - 100)
-		};
+		if (!o?.accessToken) {
+			let oAuthBackend = this.constructor.getOAuthBackend();
 
-		popup.top = (screen.height - popup.height)/2;
-		popup.left = (screen.width - popup.width)/2;
+			// Show window
+			let popup = {
+				width: Math.min(1000, innerWidth - 100),
+				height: Math.min(800, innerHeight - 100)
+			};
 
-		let state = {
-			url: location.href,
-			backend: oAuthBackend.name
-		};
+			popup.top = (screen.height - popup.height)/2;
+			popup.left = (screen.width - popup.width)/2;
 
-		this.authPopup = open(`${this.constructor.oAuth}?client_id=${this.clientId}&state=${encodeURIComponent(JSON.stringify(state))}` + this.oAuthParams(),
-			"popup", `width=${popup.width},height=${popup.height},left=${popup.left},top=${popup.top}`);
+			let state = {
+				url: location.href,
+				backend: oAuthBackend.name
+			};
 
-		if (!this.authPopup) {
-			throw new Error(this.constructor.phrase("popup_blocked"));
-		}
+			this.authPopup = open(`${this.constructor.oAuth}?client_id=${this.clientId}&state=${encodeURIComponent(JSON.stringify(state))}` + this.oAuthParams(),
+				"popup", `width=${popup.width},height=${popup.height},left=${popup.left},top=${popup.top}`);
 
-		let accessToken = await new Promise((resolve, reject) => {
-			addEventListener("message", evt => {
-				if (evt.source === this.authPopup) {
-					if (evt.data.backend == oAuthBackend.name) {
-						resolve(evt.data.token);
-					}
+			if (!this.authPopup) {
+				throw new Error(this.constructor.phrase("popup_blocked"));
+			}
 
-					if (!this.accessToken) {
-						reject(Error(this.constructor.phrase("authentication_error")));
+			accessToken = await new Promise((resolve, reject) => {
+				addEventListener("message", evt => {
+					if (evt.source === this.authPopup) {
+						if (evt.data.backend == oAuthBackend.name) {
+							resolve(evt.data.token);
+						}
+
+						if (!this.accessToken) {
+							reject(Error(this.constructor.phrase("authentication_error")));
+						}
 					}
-				}
+				});
 			});
-		});
+		}
 
 		this.accessToken = localStorage[this.constructor.tokenKey] = accessToken;
```

Comment:
Before performing this, we should probably have something like `accessToken = o.accessToken` somewhere above so we could have an access token when it's provided.

Shouldn't we have something like this?

```js
if (!o?.accessToken) {
	// Obtain an access token via a popup...
}
else {
	accessToken = o.accessToken;
}

this.accessToken = localStorage[this.constructor.tokenKey] = accessToken;
```

Reply:
LOL yes, this is what I get for submitting code I haven't tested 🙃 Do you want to post a suggestion so I can accept it?

Reply:
Done! :)